app/src/main/java/com/teamltt/carcare/DemoActivity.java

```diff
@@ -149,114 +239,53 @@ private void obdDeviceObtained() {
-                            while (true) {
-                                try {
-                                    int numRead = inputStream.read(buffer);
-                                    Log.i("are we lifting", "try");
-                                    publishProgress(buffer);
-                                } catch (IOException e) {
-                                    break;
-                                }
-                            }
-                            return null;
-                        }
-
-                        @Override
-                        protected void onProgressUpdate(byte[]... values) {
-                            Log.i("are we lifting", "try2");
-
-                            readResponse(values[0]);
-                        }
-
-                        @Override
-                        protected void onPostExecute(Void result) {
-                            connected = false;
-                            connectButton.setText("Reestablish Connection");
-                            try {
-                                inputStream.close();
-                                outputStream.close();
-                                socket.close();
-                            } catch (IOException e) {
-                                e.printStackTrace();
-                            }
-                        }
-
-                    };
-
-                    if (socket.isConnected()) {
-                        displayConnected();
-                        communicateTask.execute();
-                        Log.i("communicating", "test");
-                    } else {
-                        connectButton.setText("Retry Connect");
-                    }
-                    connecting = false;
-                }
-            };
             connectTask.execute();
 
         } catch (IOException e) {
             e.printStackTrace();
         }
     }
 
-    public void setUuid(View view) {
-        if ((findViewById(view.getLabelFor())) != null) {
-            try {
-                uuid = UUID.fromString(((TextView) findViewById(view.getLabelFor())).getText().toString());
-            } catch (IllegalArgumentException e) {
-                TextView uuidInput = (TextView) findViewById(view.getLabelFor());
-                String invalid = " - Invalid Format";
-                uuidInput.setText(uuidInput.getText().toString().replace(invalid, "") + invalid);
-            }
-        }
-    }
-
-    public void setUuidElm(View view) {
-        if ((findViewById(view.getLabelFor())) != null) {
-            uuid = UUID.fromString("00001101-0000-1000-8000-00805F9B34FB");
-        }
-    }
-
-    public void sendRequest(View view) {
+    /**
+     * Takes the text from the custom request field, appends \r to it, and writes it to the socket's output stream
+     * @param view onClick button
+     */
+    public void sendCustomRequest(View view) {
```

Comment:
No, this is an onClick function, so if the function is called (correctly) then its view parameter must exist. More importantly, it is never accessed, but it is required in the method signature by Android.